### PR TITLE
Enable full node build matrix using sudo:false

### PR DESCRIPTION
**Pull request by TheMarex**
# Issue

This PR addresses https://github.com/Project-OSRM/osrm-backend/issues/3878 by adding a build matrix using `sudo:false` jobs that skip running the cucumber tests.

## Tasklist
 - [x] review
 - [x] adjust for comments

## Requirements / Relations
Based on https://github.com/Project-OSRM/osrm-backend/pull/3914

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -132,20 +132,72 @@ matrix:
...
   - if [[ ! -z $TARGET_ARCH ]] ; then source ./scripts/travis/before_install.$TARGET_ARCH.sh ; fi
   - |
-    if [[ "${TRAVIS_OS_NAME}" == "linux" ]]; then
-      export JOBS=$((`nproc` + 1))
+    if [[ -z $JOBS ]]; then
+      if [[ "${TRAVIS_OS_NAME}" == "linux" ]]; then
+        export JOBS=$((`nproc` + 1))
+      elif [[ "${TRAVIS_OS_NAME}" == "osx" ]]; then
+        export JOBS=$((`sysctl -n hw.ncpu` + 1))
+      fi
```
Comment (daniel-j-h) [inline .travis.yml]: Can we hardcode jobs to 2 or 3, otherwise this will bite us on containerized envs.

https://docs.travis-ci.com/user/ci-environment/

Comment (TheMarex) [inline .travis.yml]: Yes there is already an override for container builds. Note the `-z $JOBS`

**Comment by TheMarex** [comment]
@daniel-j-h I would like to avoid looping over build configurations somewhere.

Using containers instead of `sudo:true` instances hopefully causes no congestion in practice, as the build time on `node-osrm` was usually below 4 minutes per job (compared to 20 minutes for our longest job right now). 

I'm not sure we should expand this to OSX builds though. Capacity there is notoriously limited and only constraining this to node 4, node 6 in release should be fine IMHO.

File: scripts/timer.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env node
+
+var exec = require('child_process').exec;
+var fs = require('fs');
+
+var name = process.argv[2];
+var cmd = process.argv.slice(3).join(' ');
+var start = +new Date();
```
Comment (danpat) [inline scripts/timer.js]: I would prefer `Date.now()` for readability instead of depending on type conversion using `+`.  Don't make me think so hard :-P

**Comment by danpat** [comment]
I'm 👍 on not building debug binaries for OSX - for debugging, I (and most OSX users I know) will build locally if we need debug.  Linux debug builds are useful for replicating production problems, but I agree, we can avoid the OSX queue on Travis here.

It's simple enough to turn on later if we need it.

**Pull request merged**